orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/PipelineExpressionEvaluator.java

```diff
@@ -97,4 +103,21 @@ public PipelineExpressionEvaluator(List<ExpressionFunctionProvider> expressionFu
 
         return expression;
       };
+
+  private void initExecutionAwareFunctions(
+      List<ExpressionFunctionProvider> expressionFunctionProviders) {
+
+    expressionFunctionProviders.forEach(
+        p -> {
+          p.getFunctions()
+              .getFunctionsDefinitions()
+              .forEach(
+                  f -> {
+                    if (!f.getParameters().isEmpty()
+                        && f.getParameters().get(0).getType() == Execution.class) {
```

Comment:
now that i look at this, i am curious if `namespace`d functions work.. .probably not (but that's a PR for another day)